### PR TITLE
Proposal form errors

### PR DESCRIPTION
**Pull request by stuartjennings-up**
Fixes for issues described here https://trello.com/c/t7tHfVxt/104-proposal-errors

### CODE REVIEW

File: src/submission/views.py [hunk 1]
```diff
@@ -372,38 +372,49 @@ def delete_incomplete_proposal(request,proposal_id):
...
-	default_fields_validated = manager_forms.DefaultForm(initial={'title': incomplete_proposal.title,'author':incomplete_proposal.author,'subtitle':incomplete_proposal.subtitle})
-	default_fields = manager_forms.DefaultForm(initial={'title': incomplete_proposal.title,'author':incomplete_proposal.author,'subtitle':incomplete_proposal.subtitle})
-
-	intial_data={}
+	default_fields_validated = manager_forms.DefaultForm(initial={'title': incomplete_proposal.title,
+																  'author':incomplete_proposal.author,
+																  'subtitle':incomplete_proposal.subtitle})
+	default_fields = manager_forms.DefaultForm(initial={'title': incomplete_proposal.title,
+														'author':incomplete_proposal.author,
+														'subtitle':incomplete_proposal.subtitle})
```
Comment (craiga) [inline src/submission/views.py]: Whoa … what's with the crazy indentation here?

Comment (craiga) [reply src/submission/views.py]: Tabs vs. spaces?

File: src/submission/views.py [hunk 2]
```diff
@@ -372,38 +372,49 @@ def delete_incomplete_proposal(request,proposal_id):
 @login_required
 def incomplete_proposal(request,proposal_id):
 	proposal_form_id = core_models.Setting.objects.get(name='proposal_form').value
-	proposal_form = manager_forms.GeneratedNotRequiredForm(form=core_models.ProposalForm.objects.get(pk=proposal_form_id))
-	default_fields = manager_forms.DefaultNotRequiredForm()
+	proposal_form = manager_forms.GeneratedForm(form=core_models.ProposalForm.objects.get(pk=proposal_form_id))
```
Comment (craiga) [inline src/submission/views.py]: We have a model called `ProposalForm`?

File: src/templates/core/proposals/assign/start_proposal.html [hunk 3]
```diff
@@ -39,25 +39,30 @@
         {% bootstrap_form default_fields %}
         {% if proposal_form %}
           <div class="row">
-         {% for field in proposal_form %}
-            <div class="{% if field.field.widget.attrs.div_class %}{{ field.field.widget.attrs.div_class }}{% else %}col-md-12{% endif %}">
+          {% for field in proposal_form %}
+            <div class="{% if field.field.widget.attrs.div_class %}{{ field.field.widget.attrs.div_class }}
+            {% else %}col-md-12{% endif %}">
```
Comment (craiga) [inline src/templates/core/proposals/assign/start_proposal.html]: These could be simplified to `{{ field.field.widget.attrs.div_class|default:"col-md-12" }}`.

File: src/templates/core/proposals/assign/start_proposal.html [hunk 4]
```diff
@@ -39,25 +39,30 @@
...
-          {% endif %}
-        </div>
+        {% endif %}
+      </div>
 
         
 	    {% buttons %}
       <div class="dash-container">
-	        <button name="book" type="submit" class="btn btn-other pull-right" style="width:120px">
+	        <button name="book" type="submit" formnovalidate class="btn btn-other pull-right" style="width:120px">
```
Comment (craiga) [inline src/templates/core/proposals/assign/start_proposal.html:64]: Why `formnovalidate`?

File: src/templates/core/proposals/assign/start_proposal.html [hunk 5]
```diff
@@ -75,7 +80,15 @@
...
     <tbody>
     <tr>
-      <td>{{ proposal.requestor.username }} <sup> <a href="core/email.html" onclick="return popitup('/email/proposal/{{proposal.id}}/user/{{proposal.requestor.id}}/','proposal_{{proposal.requestor.id}}')"class="fa fa-envelope-o btn btn-xs btn-default"></a></sup></td>
+      <td>{{ proposal.requestor.username }}
+          <sup>
+              <a href="core/email.html"
+                 onclick="return popitup('/email/proposal/{{proposal.id}}/user/{{proposal.requestor.id}}/','proposal_{{proposal.requestor.id}}')"
+                 class="fa fa-envelope-o btn btn-xs btn-default">
+
+              </a>
```
Comment (craiga) [inline src/templates/core/proposals/assign/start_proposal.html:88]: Good reformatting, but what the heck is this doing? Should the `href="core/email.html"` be there? Is there a reason we're not using the [`url`](https://docs.djangoproject.com/en/1.10/ref/templates/builtins/#url) tag? And why doesn't the link have some content (we could use the Bootstrap class `sr-only` or `text-hide` to hide it).

File: src/templates/editor/catalog/catalog.html [hunk 6]
```diff
@@ -283,7 +283,7 @@ <h4 id="files">Editors</h4>
                   <form method="POST" action="?cover=true" enctype="multipart/form-data">
                     {% csrf_token %}
                     {% bootstrap_field cover_form.cover %}
-                    <button class="btn btn-success" type="submt">Submit</button>
+                    <button class="btn btn-success" type="submit">Submit</button>
```
Comment (craiga) [inline src/templates/editor/catalog/catalog.html:286]: I guess this site doesn't see many old versions of Internet Explorer ;)

File: src/templates/submission/start_proposal.html [hunk 7]
```diff
@@ -6,146 +6,194 @@
...
+            </thead>
+            <tbody>
+            <tr>
+                <td>{{ proposal.requestor.username }}
+                    <sup>
+                        <a href="core/email.html"
+                           onclick="return popitup('/email/proposal/{{proposal.id}}/user/{{proposal.requestor.id}}/','proposal_{{proposal.requestor.id}}')"
+                           class="fa fa-envelope-o btn btn-xs btn-default">
+                        </a>
+                    </sup>
```
Comment (craiga) [inline src/templates/submission/start_proposal.html:129]: Is this repeated from the previous file? It should go into its own template. [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself).

File: src/templates/submission/start_proposal.html [hunk 8]
```diff
@@ -6,146 +6,194 @@
...
+            {% endfor %}
+        </ul>
+    {% endif %}
+    <br>
+</div>
 {% endblock body %}
 
 {% block js %}
-<script type="text/javascript" src="{% static "js/fancyfile.js" %}"></script>
+<script type="text/javascript" src="{% static " js/fancyfile.js" %}"></script>
```
Comment (craiga) [inline src/templates/submission/start_proposal.html]: Should this space be there?

File: src/templates/core/proposals/assign/start_proposal.html (hunk 4 above)
Comment (stuartjennings-up) [reply src/templates/core/proposals/assign/start_proposal.html:64]: The way Chrome interacts with HTML5 means that it automatically adds 'Please fill out this field' tags to required fields that messed with our existing validation. These tags disable that.

File: src/submission/views.py [hunk 9]
```diff
@@ -32,980 +32,980 @@
...
-	}
-
-	return render(request, template, context)
+    proposal_form_id = core_models.Setting.objects.get(name='proposal_form').value
+    proposal_form = manager_forms.GeneratedForm(form=core_models.ProposalForm.objects.get(pk=proposal_form_id))
+    default_fields = manager_forms.DefaultForm()
+    incomplete_proposal = get_object_or_404(submission_models.IncompleteProposal, pk=proposal_id)
+    proposal_form_validated = manager_forms.GeneratedForm(form=core_models.ProposalForm.objects.get(pk=proposal_form_id))
+
+    default_fields_validated = manager_forms.DefaultForm(initial={'title': incomplete_proposal.title,
```
Comment (stuartjennings-up) [inline src/submission/views.py:380]: Changing to spaces fixed this indentation

**Review by craiga — APPROVED** [review]

**Pull request merged**